src/modules/import/wmtsgetcap.js

```diff
@@ -0,0 +1,145 @@
+        };
+        layer['sourceConfig'] = ol.source.WMTS.optionsFromCapabilities(getCap, layerOptions);
+        layer['attribution'] = getCap['ServiceProvider']['ProviderName'];
+        layer['attributionUrl'] = getCap['ServiceProvider']['ProviderSite'];
+        layer['capabilitiesUrl'] = getCap['OperationsMetadata']
+          .GetCapabilities
+          .DCP
+          .HTTP
+          .Get[0]
+          .href;
+      }
+
+      layers.push(layer);
+    }
+
+    return layers;
+  };
+
+  return {
+    restrict: 'A',
+    templateUrl: ngeoWmtsGetCapTemplateUrl,
+    scope: {
+      'getCap': '=ngeoWmtsGetCap',
+      'map': '=ngeoWmtsGetCapMap',
+      'options': '=ngeoWmtsGetCapOptions'
+    },
+    link(scope) {
+
+      // List of layers available in the GetCapabilities.
+      // The layerXXXX properties use layer objects from the parsing of
+      // a  GetCapabilities file, not ol layer object.
+      scope['layers'] = [];
+      scope['options'] = scope['options'] || {};
+      scope.$watch('getCap', (val) => {
+        let err;
+        try {
+          val = new ol.format.WMTSCapabilities().read(val);
+        } catch (e) {
+          err = e;
+        }
+
+        if (err || !val) {
+          $window.console.error('WMTS GetCap parsing failed: ', err || val);
+          scope['userMsg'] = gettext('Parsing failed');
+          return;
+        }
+
+        scope['layers'] = [];
+        scope['options'].layerSelected = null; // the layer selected on click
+        scope['options'].layerHovered = null;
+
+        if (val && val['Contents'] && val['Contents']['Layer']) {
+          scope['layers'] = getLayersList(val);
+        }
+      });
+
+      // Add the selected layer to the map
+      scope['addLayerSelected'] = function() {
+        const getCapLay = scope['options'].layerSelected;
+        if (getCapLay && scope['options']['getOlLayerFromGetCapLayer']) {
+          let msg = gettextCatalog.getString('WMTS layer added succesfully');
+          try {
+            const olLayer = scope['options']['getOlLayerFromGetCapLayer'](getCapLay);
+            if (olLayer) {
+              scope['map'].addLayer(olLayer);
+            }
+
+          } catch (e) {
+            $window.console.error(`Add layer failed:${e}`);
+            msg = `${gettextCatalog.getString('WMTS layer could not be added')} ${e.message}`;
+          }
+          $window.alert(msg);
+        }
+      };
+
+      // Get the abstract to display in the text area
+      scope['getAbstract'] = function() {
+        const l = scope['options'].layerSelected || scope['options'].layerHovered || {};
+        return gettextCatalog.getString(l.Abstract) || '';
+      };
+    }
+  };
+};
+
+exports.module = angular.module('ngeo.wmtsGetCapDirective', [
+  'gettext'
+]);
+
+exports.module.value('ngeoWmtsGetCapTemplateUrl',
+    /**
+     * @param {angular.JQLite} element Element.
+     * @param {angular.Attributes} attrs Attributes.
+     * @return {boolean} Template URL.
```

Comment:
True. I made a copy paste from [wmsgetcap](https://github.com/camptocamp/ngeo/blob/master/src/modules/import/wmsgetcap.js#L201) where it says boolean. Should I correct the type there as well?